server/autolinker.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"errors"
+	"regexp"
+)
+
+// AutoLinker helper for replace regex with links
+type AutoLinker struct {
+	link    *Link
+	pattern *regexp.Regexp
+}
+
+// NewAutoLinker create and initialize a AutoLinker
+func NewAutoLinker(link *Link) (*AutoLinker, error) {
+	if link == nil || len(link.Pattern) == 0 || len(link.Template) == 0 {
+		return nil, errors.New("Pattern or template was empty")
+	}
+
+	if !link.DisableNonWordPrefix {
+		link.Pattern = "(?P<MMDisableNonWordPrefix>^|\\s)" + link.Pattern
```

Comment:
s/MMDisableNonWordPrefix/DisableNonWordPrefix/ to maintain consistency with Field names in Link Struct